app/src/main/java/ph/bohol/dictionaryapp/DictionaryDatabase.java

```diff
@@ -187,7 +187,9 @@ Cursor getEntry(final int entryId) {
 
     private String getEntryHtml(final int entryId) {
         Cursor cursor = getEntry(entryId);
-        String entryXml = cursor.getString(cursor.getColumnIndex(DictionaryDatabase.ENTRY_ENTRY));
+        int entryIndex = cursor.getColumnIndex(DictionaryDatabase.ENTRY_ENTRY);
+        if (entryIndex < 0) return ""; // This is an error
```

Comment:
May some exception handling would be nicer here (although I myself made it even more quick-and-dirty).

Reply:
Android Studio gave a red underline, which made me think it was a compiler error (the params are annotated with their legal `int` range). But it compiles fine without the range check, so I really don't know what you'd like to do. I guess crashing the app is sensible, since it means the database or the database connection is corrupted.